src/main/java/com/redhat/devtools/lsp4ij/DocumentContentSynchronizer.java

```diff
@@ -74,34 +60,66 @@ public DocumentContentSynchronizer(@NotNull LanguageServerWrapper languageServer
         textDocument.setText(document.getText());
 
-        String languageId = getLanguageId(document, languageServerWrapper);
+        @NotNull String languageId = getLanguageId(document, languageServerWrapper);
         textDocument.setLanguageId(languageId);
         textDocument.setVersion(++version);
-        didOpenFuture = languageServerWrapper.getInitializedServer()
-                .thenAcceptAsync(ls -> ls.getTextDocumentService().didOpen(new DidOpenTextDocumentParams(textDocument)));
+        didOpenFuture = languageServerWrapper
+                .getInitializedServer()
+                .thenAcceptAsync(ls -> ls.getTextDocumentService()
+                        .didOpen(new DidOpenTextDocumentParams(textDocument)));
 
         // Initialize LSP change events
         changeEvents = new ArrayList<>();
     }
 
-    private static String getLanguageId(Document document, LanguageServerWrapper languageServer) {
+    /**
+     * Returns the LSP language id defined in mapping otherwise the {@link Language#getID()} otherwise the {@link FileType#getName()} otherwise "UNKNOWN".
+     *
+     * @param document       the document.
+     * @param languageServer the language server.
+     * @return the LSP language id.
+     */
+    private static @NotNull String getLanguageId(@NotNull Document document, @NotNull LanguageServerWrapper languageServer) {
         VirtualFile file = FileDocumentManager.getInstance().getFile(document);
         if (file == null) {
-            return null;
+            return FileTypes.UNKNOWN.getName();
         }
-        Language language = LSPIJUtils.getFileLanguage(file, languageServer.getProject());
-        if (language != null) {
-            String languageId = languageServer.getLanguageId(language);
-            if (languageId != null) {
-                return languageId;
-            }
+
+        Project project = languageServer.getProject();
+
+        // 1. Try to get the LSP languageId by using language mapping
+        Language language = project != null ? LSPIJUtils.getFileLanguage(file, project) : null;
+        String languageId = languageServer.getLanguageId(language);
+        if (languageId != null) {
+            return languageId;
         }
+
+        // 2. Try to get the LSP languageId by using the fileType mapping
         FileType fileType = file.getFileType();
-        return languageServer.getLanguageId(fileType);
+        languageId = languageServer.getLanguageId(fileType);
+        if (languageId != null) {
+            return languageId;
+        }
+
+        // 3. Try to get the LSP languageId by using the file name pattern mapping
+        languageId = languageServer.getLanguageId(file.getName());
+        if (languageId != null) {
+            return languageId;
+        }
+
+        // At this step it exists no mapping
+
+        if (language != null) {
+            // The language exists, use her ID
+            return language.getID();
```

Comment:
language.getID().toLowerCase()

Reply:
You want to do that to try to map languageIds whi h are specfied in lsp?

If we do that we need to do the same thing with filetype